spec/lib/danger/danger_core/plugins/dangerfile_git_plugin_spec.rb

```diff
@@ -99,6 +99,27 @@ module Danger
           end
         end
       end
+
+      describe "getting stats for a specific file" do
+        it "returns nil when specific stats do not exist" do
+          run_in_repo_with_diff do |git|
+            diff = git.diff("master")
+            allow(@repo).to receive(:diff).and_return(diff)
```

Comment:
On second thoughts: I think we do still need the stubbing, or at least, removing it is beyond the shallow understanding I have so far. It's giving us access to a pre-prepared diff (see `run_in_repo_with_diff` at the head of the file) which we otherwise wouldn't have because `@repo` doesn't really have a diff to report on.